sns_ik_lib/src/sns_acceleration_ik.cpp

```diff
@@ -0,0 +1,190 @@
+/*! \file sns_acceleration_ik.cpp
+ * \brief Basic SNS acceleration IK solver
+ * \author Andy Park
+ */
+/*
+ *    Copyright 2018 Rethink Robotics
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <sns_ik/sns_acceleration_ik.hpp>
+
+#include <ros/console.h>
+
+#include "sns_ik_math_utils.hpp"
+
+namespace sns_ik {
+
+static const Eigen::IOFormat EigArrFmt(4, 0, ", ", "\n", "[", "]");
+
+SNSAccelerationIK::SNSAccelerationIK(int dof, double loop_period) :
+  n_dof(0),
+  n_tasks(0),
+  m_usePositionLimits(true)
+{
+  setNumberOfDOF(dof);
+  setLoopPeriod(loop_period);
+
+  baseIkSolver = SnsAccIkBase::create(n_dof);
+}
+
+void SNSAccelerationIK::setNumberOfDOF(int dof)
+{
+  if (dof > 0 && dof != n_dof) {
+    n_dof = dof;
+  }
+}
+
+void SNSAccelerationIK::setNumberOfTasks(int ntasks, int dof)
+{
+  setNumberOfDOF(dof);
+
+  if (n_tasks != ntasks) {
+    n_tasks = ntasks;
+    Eigen::VectorXd ddq = Eigen::VectorXd::Zero(n_dof);
```

Comment:
unnecessary? Or is this supposed to set a member variable?

Reply:
It is a member variable. I followed the original code's convention, so it is hard to tell whether it is a local variable or not.